tools/image/main.go

```diff
@@ -242,3 +274,18 @@ func lifecycleLayer() string {
 	}
 	return lf.Name()
 }
+
+func pullImage(dockerCli dockercli.CommonAPIClient, ref string) error {
+	rc, err := dockerCli.ImagePull(context.Background(), ref, dockertypes.ImagePullOptions{})
+	if err != nil {
+		// Retry
+		rc, err = dockerCli.ImagePull(context.Background(), ref, dockertypes.ImagePullOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	if _, err := io.Copy(ioutil.Discard, rc); err != nil {
+		return err
+	}
+	return rc.Close()
```

Comment:
I think we want to `defer rc.Close()` before we copy. If there is an error during copy the reader will not be closed properly.